File: Examples/Access/C#/NetOffice COMAddin Examples/02 Ribbons And Panes/SamplePane.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using System.Diagnostics;

namespace Access02AddinCS4
{
    public partial class SamplePane : UserControl, NetOffice.AccessApi.Tools.ITaskPane // Not necessary to implement ITaskPane but its helpful
    {
        #region Ctor

        public SamplePane()
        {           
            InitializeComponent();
        }

        #endregion

        #region Properties

        private PerformanceCounter Counter { get; set; }

        #endregion

        #region ITaskpane

        public void OnConnection(NetOffice.AccessApi.Application application, NetOffice.OfficeApi._CustomTaskPane parentPane, object[] customArguments)
        {

        }

        public void OnDisconnection()
        {
            UsageTimer.Enabled = false;
            if (null != Counter)
            {
                Counter.Dispose();
                Counter = null;
            }
        }

        public void OnVisibleStateChanged(bool visible)
        {
            // Create the performance counter is expensive in performance
            // To avoid slow down the Access startup sequence - we create them on demand when user want show the pane
            // (Real world code want doing that async)
            if (visible && null == Counter)
            {
                Counter = new PerformanceCounter("Process", "% Processor Time", "MSACCESS");
                UsageTimer.Enabled = true;
            }
            else if (visible)
                UsageTimer.Enabled = true;
            else if (!visible)
                UsageTimer.Enabled = false;
        }

        public void OnDockPositionChanged(NetOffice.OfficeApi.Enums.MsoCTPDockPosition position)
        {
            
        }

        #endregion

        #region UI Trigger

        protected override void OnResize(EventArgs e)
        {
            base.OnResize(e);
            UsageLabel.Location = new Point(
                                    (Width / 2 - UsageLabel.Width / 2),
                                    (Height / 2 - UsageLabel.Height / 2));
        }

        private void UsageTimer_Tick(object sender, EventArgs e)
        {
            if (null != Counter)
            {
                float value = Counter.NextValue();
              
                int barValue = Convert.ToInt32(value);
                if (barValue < 0)
                    barValue = 0;
                if (barValue > 100)
                    barValue = 100;
                UsageLabel.Text = String.Format("{0} %", barValue);
                UsageBar.Value = barValue;
            }
            else
            {
                UsageLabel.Text = String.Empty;
            }
        }

        #endregion
    }
}

File: Examples/Excel/VB/NetOffice COMAddin Examples/02 Ribbons and Panes/SamplePane.vb
Imports NetOffice.ExcelApi.Tools
Imports System.Drawing

Public Class SamplePane
    Implements ITaskPane ' Not necessary to implement ITaskPane but its helpful

    Private Counter As PerformanceCounter

    Public Sub OnConnection(ByVal application As NetOffice.ExcelApi.Application, ByVal parentPane As NetOffice.OfficeApi._CustomTaskPane, ByVal customArguments() As Object) Implements NetOffice.ExcelApi.Tools.ITaskPane.OnConnection


    End Sub

    Public Sub OnDisconnection() Implements NetOffice.ExcelApi.Tools.ITaskPane.OnDisconnection

        UsageTimer.Enabled = False
        If Not IsNothing(Counter) Then
            Counter.Dispose()
            Counter = Nothing
        End If

    End Sub

    Public Sub OnVisibleStateChanged(ByVal visible As Boolean) Implements NetOffice.ExcelApi.Tools.ITaskPane.OnVisibleStateChanged

        ' Create the performance counter is expensive in performance
        ' To avoid slow down the Excel startup sequence - we create them on demand when user want show the pane
        ' (Real world code want doing that async)
        If (visible And IsNothing(Counter)) Then
            Counter = New PerformanceCounter("Process", "% Processor Time", "Excel")
            UsageTimer.Enabled = True
        ElseIf True = visible Then
            UsageTimer.Enabled = True
        ElseIf False = visible Then
            UsageTimer.Enabled = False
        End If

    End Sub

    Public Sub OnDockPositionChanged(ByVal position As NetOffice.OfficeApi.Enums.MsoCTPDockPosition) Implements NetOffice.ExcelApi.Tools.ITaskPane.OnDockPositionChanged


    End Sub

    Protected Overrides Sub OnResize(e As EventArgs)

        MyBase.OnResize(e)
        UsageLabel.Location = New Point(
                                    (Width / 2 - UsageLabel.Width / 2),
                                    (Height / 2 - UsageLabel.Height / 2))

    End Sub

    Private Sub UsageTimer_Tick(sender As Object, e As EventArgs) Handles UsageTimer.Tick

        If Not IsNothing(Counter) Then

            Dim value As Single = Counter.NextValue()
            Dim barValue As Int32 = Convert.ToInt32(value)
            If (barValue < 0) Then barValue = 0
            If (barValue > 100) Then barValue = 100
            UsageLabel.Text = String.Format("{0} %", barValue)
            UsageBar.Value = barValue

        Else

            UsageLabel.Text = String.Empty

        End If

    End Sub

End Class